app/services/shelf_list.rb

```diff
@@ -50,7 +58,7 @@ def reverse_keys
       @reverse_keys ||= TermsQuery.call(
         field: reverse_shelfkey,
         limit: reverse_limit,
-        query: ShelfKey.reverse(query)
+        query: ShelfKey.reverse(query, classification)
```

Comment:
Technically, we don't _have_ to know the classification in order to reverse it. We can just reverse whatever the query is. We'd need to have another method for "plain reverse" or re-think the implementation.

Reply:
I got rid of classification from ShelfKey and moved it the prefix check to ShelfList, pls let me know what you think of that.